### ISSUE TITLE
Pill sometimes confused of displayname

### ISSUE THREAD

**Issue by eras** [vector-im/riotX-android#1046]
Setting different displayname per room can sometimes confuse Riot-X what the correct displayname for a particular room is.

![Captured from RiotX](https://user-images.githubusercontent.com/1159374/75029851-8ffbb500-54ab-11ea-917e-a54e7f5b1523.png)
![Captured from RiotWeb](https://user-images.githubusercontent.com/1159374/75029879-9e49d100-54ab-11ea-9dab-43e7c656b0ba.png)

The room's displayname is "flux". However, my displayname on another room is "erkki" and I had recently received messages on that room. So it sounds like the displayname is cached per mxid, not per room+mxid.

On a later time I tried to view the same message I got the correct pill "flux".

**Comment by bmarty** [comment]
Yes, pills in the timeline use user data and not room member data (which can differ between rooms)

Ref: https://github.com/vector-im/riotX-android/blob/54f2ac0d8cea552e71b4691a0089c37e9772d9b6/vector/src/main/java/im/vector/riotx/features/html/MxLinkTagHandler.kt#L45

We will improve this in the future
